mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1272,9 +1314,22 @@ static FuncOp launchGPUConvolution(ModuleOp &module, OpBuilder &builder,
   } else if (dataType == builder.getIntegerType(16)) {
     gpuMemDeallocFuncName = "mgpuMemDealloc5DBF16";
   }
-  auto mgpuMemDealloc5DFuncOp = makeFuncDecl(
-      builder, gpuMemDeallocFuncName, {fiveDimUnknownSizeMemRefType}, {});
-  module.push_back(mgpuMemDealloc5DFuncOp);
+
+  FuncOp mgpuMemDealloc5DFuncOp;
+  bool deallocFuncExist = false;
+  module.walk([&](FuncOp funcOp) -> WalkResult {
+    if (funcOp.getName() == gpuMemDeallocFuncName) {
+      mgpuMemDealloc5DFuncOp = funcOp;
+      deallocFuncExist = true;
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+  if (!deallocFuncExist) {
```

Comment:
Instead of having this boolean, you could instead use `wasInterrupted()` per https://mlir.llvm.org/docs/Tutorials/UnderstandingTheIRStructure/#walkers